### DISCUSSION TITLE
Behavior of model_class in Version 3.0.0a1

### DISCUSSION THREAD

**Discussion by jrast** [pallets-eco/flask-sqlalchemy#1101] [Q&A] [answered]
So far I used the following code to set the `modeL_class`:

````python
from flask_security import SQLAlchemyUserDatastore
from flask_sqlalchemy import SQLAlchemy
from sqlalchemy import Column, Integer
from sqlalchemy.ext.declarative import declared_attr, declarative_base


_Base = declarative_base()


class BaseModel(_Base):
    """Base model for database objects

    The base model contains a `id` column as primary key
    """

    __abstract__ = True

    @declared_attr
    def __table_args__(cls):
        return {
            "mysql_engine": "InnoDB",
        }

    id = Column(Integer, primary_key=True)


db: SQLAlchemy = SQLAlchemy(model_class=BaseModel)
````
This was working as expected and the `.query` property was added to the models as expected. Since Version 3.0.0a1 this no longer works (for example the `.query` property is missing). Is this intentional? Is it required to use `flask_sqlalchemy.Model` now as a base class? If this is the case, this would make it harder to use the models outside of the flask context which is required in some cases.

**Comment by davidism** [comment, marked as answer]
This was an oversight while rewriting the code, I'll add `model.query` for existing bases back in.

In general, it's required to subclass Flask-SQLAlchemy classes to get Flask-SQLAlchemy behaviors. Note that you're also missing out on other behaviors, such as automatic table names, bind keys, and a nice `repr`.

However, the entire `query` interface is considered legacy in SQLAlchemy now. [See the discussion in their docs.](https://docs.sqlalchemy.org/en/14/orm/tutorial.html) SQLAlchemy wants you to move to `db.select(User)` to construct queries, and `db.session.execute(q)` to execute them.

**Reply by jrast** [reply]
Would you recommend to use the `select` / `execute` interface for extensions? At least where the additional functionality of flask-sqlalchemy is not required.

**Reply by davidism** [reply]
I recommend whatever SQLAlchemy recommends. The extension changes very little about the actual use of SQLAlchemy, that hasn't changed.
